File: njob-rest/src/main/java/com/github/nsorin/njobrest/model/Company.java
package com.github.nsorin.njobrest.model;

import java.util.List;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.validation.constraints.NotNull;

import com.fasterxml.jackson.annotation.JsonManagedReference;

import org.hibernate.validator.constraints.URL;

import lombok.Getter;
import lombok.Setter;

@Entity
public class Company {

    @GeneratedValue
    @Id
    @Getter
    @Setter
    private Integer id;

    @Getter
    @Setter
    @NotNull
    private String name;

    @Getter
    @Setter
    private String address;

    @Getter
    @Setter
    private String city;

    @Getter
    @Setter
    private Integer zipCode;

    @Getter
    @Setter
    @URL
    private String webSite;

    @JsonManagedReference
    @OneToMany(mappedBy = "company")
    @Getter
    @Setter
    private List<Application> applications;
    
    @JsonManagedReference
    @OneToMany(mappedBy = "company")
    @Getter
    @Setter
    private List<Recruiter> recruiters;
}

File: njob-frontend/src/store/models/Company.js
import { Model } from "@vuex-orm/core"
import Application from "./Application"
import Recruiter from "./Recruiter"

export default class Company extends Model {

    static entity = 'company'

    static apiConfig = {
        actions: {
            fetch: {
                method: 'get',
                url: '/companies'
            }
        }
    }

    static fields() {
        return {
            id: this.number(null),
            name: this.string(''),
            address: this.string(''),
            city: this.string(''),
            zipCode: this.number(null),
            webSite: this.string(''),
            applications: this.hasMany(Application, 'company_id'),
            recruiters: this.hasMany(Recruiter, 'company_id')
        }
    }
}